### PR TITLE
python-setup: Add support for Poetry 1.2

### PR DESCRIPTION
**Pull request by RasmusWL**
Fixes https://github.com/github/codeql-action/issues/1241

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: python-setup/auto_install_packages.py [hunk 1]
```diff
@@ -9,27 +9,49 @@
...
 
+    # only pass `env` argument if we need to pass in an updated environment
+    kwargs = {}
+    if extra_env:
+        new_env = os.environ.copy()
+        new_env.update(extra_env)
+        kwargs = {"env": new_env}
 
-def _check_output(command):
+    subprocess.check_call(command, stdin=subprocess.DEVNULL, **kwargs)
```
Comment (tausbn) [inline python-setup/auto_install_packages.py:22]: This is an interesting trick. I'm not really sure whether I like it or not (but I don't feel strongly enough about it to suggest an alternative :sweat_smile:).

**Review by tausbn — DISMISSED** [review]
This looks reasonable to me. :+1:

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## [UNRELEASED]
 
 - We will soon be rolling out a feature of the CodeQL Action that stores some information used to make future runs faster in the GitHub Actions cache. Initially, this will only be enabled on JavaScript repositories, but we plan to add more languages to this soon. The new feature can be disabled by passing the `trap-caching: false` option to your workflow's `init` step, for example if you are already using the GitHub Actions cache for a different purpose and are near the storage limit for it.
+- Add support for Python automatic dependency installation with Poetry 1.2.
```
Comment (henrymercer) [inline CHANGELOG.md:6]:
```suggestion
- Add support for Python automatic dependency installation with Poetry 1.2 [#1258](https://github.com/github/codeql-action/pull/1258).
```

File: python-setup/auto_install_packages.py (hunk 1 above)
Comment (henrymercer) [reply python-setup/auto_install_packages.py:22]: One alternative I suppose would be setting the default value of `extra_env` to `{}`, then:

```py
env = os.environ.copy()
env.update(extra_env)
subprocess.check_call(command, stdin=subprocess.DEVNULL, env=env)
```

**Review by henrymercer — APPROVED** [review]
A small tweak to the changelog note and an optional comment.

File: python-setup/auto_install_packages.py (hunk 1 above)
Comment (RasmusWL) [reply python-setup/auto_install_packages.py:22]: We could. I'm not very committed to any solution, so if you want me to, I can make the change -- just let me know how strongly you feel about this @henrymercer :blush: (optional reading: [The Sliding Scale of Giving a Fuck](https://capwatkins.com/blog/the-sliding-scale-of-giving-a-fuck))

Comment (henrymercer) [reply python-setup/auto_install_packages.py:22]: I also don't feel particularly strongly — only a 5 out of 10, if I had to rate it :)

**Review by henrymercer — DISMISSED** [review]

File: python-setup/auto_install_packages.py (hunk 1 above)
Comment (RasmusWL) [reply python-setup/auto_install_packages.py]: updated the code :+1:

**Review by henrymercer — APPROVED** [review]

**Pull request merged**